sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -484,6 +484,9 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
       case If(cond, FalseLiteral, l @ Literal(null, _)) if !cond.nullable => And(Not(cond), l)
       case If(cond, TrueLiteral, l @ Literal(null, _)) if !cond.nullable => Or(cond, l)
 
+      case CaseWhen(Seq((cond, TrueLiteral)), Some(FalseLiteral)) => cond
```

Comment:
do we already optimize single-branch CASE WHEN to IF and get this optimized?

Reply:
No. https://issues.apache.org/jira/browse/SPARK-24892

Reply:
OK so this follows https://github.com/apache/spark/pull/21850#discussion_r205505495 , LGTM